### PR TITLE
[3.3.5] Core/Spells: Rogue Honor Among Thieves

### LINKED ISSUE #16511: [3.3.5][Rogue] Core/Spell: Honor Among Thieves
3.3.5 d1a185e3d5af

Rogue talent [Honor Among Thieves](http://www.wowhead.com/spell=51701/honor-among-thieves#changelog)

Tooltip

Instead of giving the combo point to the rogue with HaT, it gives the combo point to the person that got a critical.

Repro:
- Have a feral druid and a rogue with HaT in group
- Have the feral druid crit a target with Mangle. They will gain 3 combo points (1 base + 1 Primal Fury + 1 HaT) instead of 2 (1 base + 1 Primal Fury).
- The rogue will not gain any combo points on the target the feral got a crit on.

Expected behavior:
- The rogue will gain a combo point and the feral will gain two.

### PR DESCRIPTION
**Pull request by ccrs**
fix #16511

### CODE REVIEW

File: src/server/scripts/Spells/spell_rogue.cpp [hunk 1]
```diff
@@ -703,6 +706,120 @@ class spell_rog_tricks_of_the_trade_proc : public SpellScriptLoader
...
+        void HandleProc(AuraEffect const* aurEff, ProcEventInfo& eventInfo)
+        {
+            PreventDefaultAction();
+
+            Unit* caster = GetCaster();
+            if (!caster)
+                return;
+
+            Unit* target = GetTarget();
+            target->CastSpell(target, eventInfo.GetSpellInfo()->Id, true, nullptr, aurEff, caster->GetGUID());
```
Comment (joschiwald) [inline src/server/scripts/Spells/spell_rogue.cpp]: thats wrong, eventInfo.GetSpellInfo() return the spell which triggers the aura proc

File: src/server/scripts/Spells/spell_rogue.cpp [hunk 2]
```diff
@@ -703,6 +706,120 @@ class spell_rog_tricks_of_the_trade_proc : public SpellScriptLoader
...
+    {
+        PrepareAuraScript(spell_rog_honor_among_thieves_AuraScript);
+
+        bool CheckProc(ProcEventInfo& eventInfo)
+        {
+            Unit* caster = GetCaster();
+            if (!caster)
+                return false;
+
+            if (caster->GetTypeId() == TYPEID_PLAYER && !caster->GetSpellHistory()->HasCooldown(eventInfo.GetSpellInfo()->Id))
```
Comment (joschiwald) [inline src/server/scripts/Spells/spell_rogue.cpp]: same here

File: src/server/scripts/Spells/spell_rogue.cpp [hunk 3]
```diff
@@ -703,6 +706,120 @@ class spell_rog_tricks_of_the_trade_proc : public SpellScriptLoader
...
+    {
+        PrepareAuraScript(spell_rog_honor_among_thieves_AuraScript);
+
+        bool CheckProc(ProcEventInfo& eventInfo)
+        {
+            Unit* caster = GetCaster();
+            if (!caster)
+                return false;
+
+            if (caster->GetTypeId() == TYPEID_PLAYER && !caster->GetSpellHistory()->HasCooldown(GetSpellInfo()->Effects[EFFECT_0].TriggerSpell))
```
Comment (joschiwald) [inline src/server/scripts/Spells/spell_rogue.cpp]: is the player check required?

**Comment by Treeston** [comment]
```Spell `51698` Effect `Index: EFFECT_0 AuraName: 6` of script `spell_rog_honor_among_thieves` did not match dbc effect data - handler bound to hook `OnEffectProc` of AuraScript won't be executed
Spell `51700` Effect `Index: EFFECT_0 AuraName: 6` of script `spell_rog_honor_among_thieves` did not match dbc effect data - handler bound to hook `OnEffectProc` of AuraScript won't be executed
Spell `51701` Effect `Index: EFFECT_0 AuraName: 6` of script `spell_rog_honor_among_thieves` did not match dbc effect data - handler bound to hook `OnEffectProc` of AuraScript won't be executed```

**Comment by ccrs** [comment]
tested and working

File: src/server/scripts/Spells/spell_rogue.cpp [hunk 4]
```diff
@@ -703,6 +706,116 @@ class spell_rog_tricks_of_the_trade_proc : public SpellScriptLoader
...
+    }
+
+    class spell_rog_honor_among_thieves_proc_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_rog_honor_among_thieves_proc_AuraScript);
+
+        void HandleEffectApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+        {
+            if (Player* player = GetTarget()->ToPlayer())
+                if (Unit* spellTarget = ObjectAccessor::GetUnit(*player, player->GetTarget()))
```
Comment (Treeston) [inline src/server/scripts/Spells/spell_rogue.cpp]: Casts at current target even if the rogue has CP on a different target. Intended?

Comment (Treeston) [inline src/server/scripts/Spells/spell_rogue.cpp]: Blizzlike confirmed. Should be "rogue's current target _as long as the rogue is in combat with that target_".

Comment (ccrs) [inline src/server/scripts/Spells/spell_rogue.cpp]: yes

**Pull request merged**